### PR TITLE
[DOCUMENT STATUS] Add zero state for  status files zero state

### PR DESCRIPTION
**Pull request by dfitchett**
## Are you removing, renaming or moving a folder in this PR?

- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

### :warning: TeamSites :warning:

Did you change site-wide styles, platform utilities or other infrastructure?

- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary

- **Feature Enhancement**: Introduced document status empty state functionality for the Claim Status Tool Files tab to improve user experience when viewing file upload statuses (behind feature flag)
- **New Components**: Added `FilesReceived` and `FileSubmissionsInProgress` components to display uploaded files and evidence submissions in progress with proper empty states
- **Dynamic File Size Limits**: Updated file upload hints to reflect dynamic size limits with new constants for maximum file sizes
- **Feature Toggle Support**: Integrated new components with feature toggle support for controlled rollout
- **Comprehensive Testing**: Enhanced end-to-end tests with new scenarios covering various document upload states and added comprehensive unit tests for new components
- **Team**: Benefits Management Tools Team 2 - Claims Status Tool

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/120490

## Testing done

- **Old behavior**: Users viewing the Claims Status Tool Files tab would see basic file upload interface without clear status indicators for different document states
- **Steps to verify changes**:

  1. Run mock data server and cst entry point:

  ```
  yarn mock-api --responses src/applications/claims-status/tests/local-dev-mock-api/common.js
  yarn watch --env entry=claims-status
  ```

  1. Navigate to Claims Status Tool Files tab
  2. Verify that the new `FilesReceived` component displays when there are received documents
  3. Verify that the `FileSubmissionsInProgress` component displays when there are submissions in progress
  4. Verify claims:

  - [5](http://localhost:3001/track-claims/your-claims/5/files) should show zero state content "You don’t have any file submissions in progress." and "We haven’t received any files yet."
  - [6](http://localhost:3001/track-claims/your-claims/6/files) should show "Placeholder for 1 in progress items" and "We haven’t received any files yet."
  - [7](http://localhost:3001/track-claims/your-claims/7/files) should show "We’ve received all the files you’ve uploaded." and "Placeholder for 1 files received"
  - [8](http://localhost:3001/track-claims/your-claims/8/files) should show "Placeholder for 1 in progress items" and "Placeholder for 1 files received"

  5. Verify feature toggle functionality enables/disables new components appropriately
  6. Test accessibility with screen readers and keyboard navigation

- **Tests completed**:
  - Added comprehensive unit tests for new components (`FilesReceived.unit.spec.jsx`, `FileSubmissionsInProgress.unit.spec.jsx`)
  - Enhanced existing unit tests for modified components
  - Added end-to-end test scenarios in `02.claim-files.v2.cypress.spec.js` covering:
    - No evidence submissions with no supporting documents
    - No evidence submissions with supporting documents
    - One evidence submission with no supporting documents
    - One evidence submission with supporting documents
  - All existing tests continue to pass
  - Accessibility checks included in all new tests
- **Results**: All tests pass, new components render correctly in various states, feature toggles work as expected, accessibility compliance maintained

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before                                       | After                                                              |
| ------- | -------------------------------------------- | ------------------------------------------------------------------ |
| Desktop |  |  Screenshot 2025-09-29 at 11 33 34 AM |
|  |  | Screenshot 2025-09-29 at 11 38 25 AM |
|  |  | Screenshot 2025-09-29 at 11 39 12 AM |
|  |  | Screenshot 2025-09-29 at 11 39 35 AM |

## What areas of the site does it impact?

- **Primary Impact**: Claims Status Tool - Files tab (`/claim-status/files` routes)
- **Components Modified**: File upload and document display components within claims status application
- **Feature Scope**: Limited to claims status application, no cross-application impacts
- **User Experience**: Improved clarity for veterans viewing their document upload status and evidence submissions

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

**Specific areas for review**:

- **Component Architecture**: Review the new `FilesReceived` and `FileSubmissionsInProgress` components for proper separation of concerns and reusability
- **Feature Toggle Integration**: Ensure the feature toggle implementation follows platform standards and provides smooth rollout capability
- **Test Coverage**: Verify that the new test scenarios adequately cover edge cases and various document status combinations
- **Accessibility**: Confirm that the new empty state components maintain proper accessibility standards and screen reader compatibility

### CODE REVIEW

File: src/applications/claims-status/containers/FilesPage.jsx [hunk 1]
```diff
@@ -85,6 +87,12 @@ class FilesPage extends React.Component {
         <ClaimFileHeader isOpen={isOpen} />
         <Toggler toggleName={Toggler.TOGGLE_NAMES.cstShowDocumentUploadStatus}>
           <Toggler.Enabled>
+            <AdditionalEvidencePage additionalEvidenceTitle="Upload additional evidence" />
+            <div className="vads-u-margin-y--6 vads-u-border--1px vads-u-border-color--gray-light" />
```
Comment (JoshingYou1) [inline src/applications/claims-status/containers/FilesPage.jsx:91]: Do we know if the design system has something that creates these lines? If not, this is great!

**Review by JoshingYou1 — DISMISSED** [review]
LGTM! Nice work 🚀

**Review by asiisii — CHANGES_REQUESTED** [review]
Nice work on this! The tests look great, and everything worked as expected when I tested it locally.  
Just replace `moment` with `new Date()` for sorting, like you did in  
`src/applications/claims-status/components/claim-files-tab-v2/FileSubmissionsInProgress.jsx`.

Also, it looks like the image link for the `before` page in your PR description is broken @dfitchett

**Review by asiisii — APPROVED** [review]
LGTM 🚀

**Review by RyanMunsch — APPROVED** [review]
